music21/musicxml/partStaffExporter.py

```diff
@@ -773,13 +771,17 @@ def moveMeasureContents(measure: Element, otherMeasure: Element, staffNumber: in
                     otherMeasure.remove(existingBarline)
             if elem.tag == 'note':
                 voice = elem.find('voice')
+                # otherMeasure is already using voice numbers 1 through maxVoices.
+                # Adjust/set this note's voice number to avoid that range.
                 if voice is not None:
-                    if otherMeasureLackedVoice and voice.text:
-                        # otherMeasure assigned voice 1; Bump voice number here
-                        voice.text = str(int(voice.text) + 1)
+                    if voice.text:
+                        # increment current voice number by maxVoices
+                        voice.text = str(maxVoices + int(voice.text))
```

Comment:
Won't this destroy intentional cross-staff beams?

Reply:
I think it will.  Damn.

I don't believe my Humdrum parser is assigning voice numbers (but I'll check more thoroughly).  I think that voices are currently being blindly assigned voice numbers 1..N (per PartStaff) _somewhere_, and that means that partStaffExporter (unfixed) always thinks there are cross-staff voices.  My fix makes it so partStaffExporter _never_ thinks there are cross-staff voices, which is also incorrect.  I need to find who is assigning these bad voice numbers and fix it there, so I can put partStaffExporter back the way it was.

Reply:
Musescore/Finale don't care about voice numbers, so maybe we never noticed this bug when it was introduced?  Dorico is very unhappy, though.

Reply:
I think it's this:
https://github.com/cuthbertLab/music21/blob/2dccb5b34d181eb2038b37ea8d722fa1fe848b4a/music21/musicxml/m21ToXml.py#L3106

Your test case has memory locations for the id number (default), and that's nonsense in a musicxml document, so this is the part we could make part-staff-aware.

Reply:
I think you're right.  Not sure how to do that, though.  I could (thinking out loud) mark these ids as "arbitrary" somehow, so partStaffExporter knows it can overwrite them?

Reply:
You have access to the PartExporter as self.parent, and from there the part object, and `PartExporter.groupsToJoin`, so you could walk the groups, find out if the current part is not-the-first, and then advance the voice numbers. Something like that. Then no changes (hopefully) in partStaffExporter, or even some code we could remove, possibly (haven't looked).

Reply:
A model:
https://github.com/cuthbertLab/music21/blob/2dccb5b34d181eb2038b37ea8d722fa1fe848b4a/music21/musicxml/m21ToXml.py#L2728-L2729

Reply:
I'll give it a shot.  Learning, learning, learning... :-)